### PR TITLE
Eventarc Quickstart Brushup

### PR DESCRIPTION
**Pull request by grant**
## Eventarc Quickstart Brushup

- move `run/events-*` to `eventarc/`
- Region tag fixes
- Adds missing Dockerfile
- Does not change content of samples

### CODE REVIEW

**Comment by elharo** [comment]
[java] JVM args ignored when same JVM is used.
[ERROR] High: Found reliance on default encoding in com.example.cloudrun.EventController.receiveMessage(PubSubBody, Map): new String(byte[]) [com.example.cloudrun.EventController] At EventController.java:[line 53] DM_DEFAULT_ENCODING
[ERROR] Medium: Method com.example.cloudrun.EventController.receiveMessage(PubSubBody, Map) encodes String bytes without specifying the character encoding [com.example.cloudrun.EventController] At EventController.java:[line 53] MDM_STRING_BYTES_ENCODING
[ERROR] Low: com.example.cloudrun.EventController is a Spring endpoint (Controller) [com.example.cloudrun.EventController] At EventController.java:[lines 38-57] SPRING_ENDPOINT
[ERROR] Low: Method com.example.cloudrun.eventpojos.PubSubBody.getMessage() that can return null, is missing a @Nullable annotation [com.example.cloudrun.eventpojos.PubSubBody] At PubSubBody.java:[line 25] AI_ANNOTATION_ISSUES_NEEDS_NULLABLE
[ERROR] Medium: Unwritten field: com.example.cloudrun.eventpojos.PubSubBody.message [com.example.cloudrun.eventpojos.PubSubBody] At PubSubBody.java:[line 25] UWF_UNWRITTEN_FIELD
[ERROR] Medium: Should com.example.cloudrun.eventpojos.PubSubBody$PubSubMessage be a _static_ inner class? [com.example.cloudrun.eventpojos.PubSubBody$PubSubMessage] At PubSubBody.java:[lines 28-64] SIC_INNER_SHOULD_BE_STATIC
[ERROR] Failed to execute goal com.github.spotbugs:spotbugs-maven-plugin:4.0.4:check (analyze-compile) on project events-pubsub: failed with 6 bugs and 0 errors -> [Help 1]
[ERROR]

File: eventarc/pubsub/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2020 Google LLC
```
Comment (averikitsch) [inline eventarc/pubsub/Dockerfile:1]: If this isn't explicitly for a quickstart tutorial like the audit logs please remove.

File: eventarc/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+[![Open in Cloud Shell][shell_img]][shell_link]
+
+[shell_img]: http://gstatic.com/cloudssh/images/open-btn.png
+[shell_link]: https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/java-docs-samples&page=editor&open_in_editor=blog/README.md
+
+This directory contains samples for Cloud Eventarc.
+
+## Samples
+
+|           Sample                |        Description       |     Deploy    |
```
Comment (averikitsch) [inline eventarc/README.md:12]: remove deploy column if not using.

File: eventarc/pubsub/Dockerfile (hunk 1 above)
Comment (grant) [reply eventarc/pubsub/Dockerfile:1]: This is for the Eventarc Pub/Sub quickstart.

File: eventarc/README.md (hunk 2 above)
Comment (grant) [reply eventarc/README.md:12]: The deploy column should still have the 2 quickstarts there were before. I didn't change the code, just moved the samples.

File: eventarc/README.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+| ------------------------------- | ------------------------ | ------------- |
+|[Events - Pub/Sub](events-pubsub/) | Events for Cloud Run with Pub/Sub | [<img src="https://storage.googleapis.com/cloudrun/button.svg" alt="Run on Google Cloud" height="30">][run_button_events_pubsub] |
+|[Anthos Events - Pub/Sub](events-pubsub/anthos.md) | Events for Cloud Run on Anthos with Pub/Sub | - |
+|[Events - GCS](events-gcs/) | Events for Cloud Run with GCS | [<img src="https://storage.googleapis.com/cloudrun/button.svg" alt="Run on Google Cloud" height="30">][run_button_events_gcs] |
```
Comment (averikitsch) [inline eventarc/README.md:16]: Do these names need to be updated?

File: eventarc/pubsub/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+# It's important to use OpenJDK 8u191 or above that has container support enabled.
+# https://hub.docker.com/r/adoptopenjdk/openjdk8
+# https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds
+FROM adoptopenjdk/openjdk11:alpine-slim
+
+# Copy the jar to the production image from the builder stage.
+COPY --from=builder /app/target/events-storage-*.jar /events-storage.jar
+
+# Run the web service on container startup.
+CMD ["java", "-Djava.security.egd=file:/dev/./urandom", "-jar", "/events-storage.jar"]
```
Comment (averikitsch) [inline eventarc/pubsub/Dockerfile:39]:
```suggestion
COPY --from=builder /app/target/events-pubsub-*.jar /events-pubsub.jar

# Run the web service on container startup.
CMD ["java", "-Djava.security.egd=file:/dev/./urandom", "-jar", "/events-pubsub.jar"]
```

**Review by averikitsch — CHANGES_REQUESTED** [review]

File: eventarc/README.md (hunk 3 above)
Comment (grant) [reply eventarc/README.md]: Yup, fixed. Thanks.

File: eventarc/pubsub/Dockerfile (hunk 4 above)
Comment (grant) [reply eventarc/pubsub/Dockerfile]: Applied.

File: eventarc/README.md (hunk 2 above)
Comment (grant) [reply eventarc/README.md:12]: Should still be there with the buttons.

**Review by grayside — COMMENTED** [review]
README & Region tags LGTM, but is the Dockerfile needed? Aside from the Cloud Run quickstart, the default practice for Java samples in Cloud Run is to use `jib`. @averikitsch

**Comment by averikitsch** [comment]
> README & Region tags LGTM, but is the Dockerfile needed? Aside from the Cloud Run quickstart, the default practice for Java samples in Cloud Run is to use `jib`. @averikitsch

For quickstarts, we are allowing Dockerfiles in order to keep languages consistent.

File: eventarc/audit-storage/Dockerfile [hunk 5]
```diff
@@ -33,9 +33,9 @@ RUN mvn package -DskipTests
 FROM adoptopenjdk/openjdk11:alpine-slim
 
 # Copy the jar to the production image from the builder stage.
-COPY --from=builder /app/target/events-storage-*.jar /events-storage.jar
+COPY --from=builder /app/target/audit-storage-*.jar /audit-storage.jar
```
Comment (averikitsch) [inline eventarc/audit-storage/Dockerfile:36]: Sorry missed a comment: for this change the pom.xml needs the name filed changed to audit-storage as well

Comment (grant) [reply eventarc/audit-storage/Dockerfile:36]: Thanks.

**Review by averikitsch — APPROVED** [review]

**Comment by grant** [comment]
(Letting Averi take this PR)

**Review by averikitsch — APPROVED** [review]

**Comment by averikitsch** [comment]
Repacking step added for Dockerfile support.

**Comment by grant** [comment]
Thanks @averikitsch!

**Pull request merged**